core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+      header <- headers.map { case (k, v) => k.toLowerCase -> v }.collect { case ("link", header) => header }
+      LinkRelation(url, relation) <- LinkRelation.findAllMatchIn(header)
+      uri <- Uri.fromString(url).toSeq
+      page <- uri.params.get("page")
+      pageNum <- Try(page.toInt).toOption
+      perPage <- uri.params.get("per_page")
+      perPageNum <- Try(perPage.toInt).toOption
+    } yield (relation, (pageNum, perPageNum))
+    relations.toMap
+  }
+
+  /** Extract change label from trickleUpdateDependencies log */
+  def extractLabel(lines: List[String]): ChangeLabel = {
+    lines.collectFirst {
+      case ChangeLabel(label) => label
+    }.getOrElse(sys.error("Change label not found!"))
+  }
+
+  /** Extract updated versions from trickleUpdateDependencies log */
+  def extractChanges(lines: List[String]): List[String] = {
+    lines.filter(_ contains "Updated ").map(line => line.substring(line.indexOf("Updated ")))
+  }
+
+  def getBranch(pullRequest: Option[PullRequestDraft]): IO[(String, String)] = IO {
+    pullRequest
+      .map("" -> _.head.get.ref)
+      .getOrElse("-b" -> s"trickle/version-bump-${System.currentTimeMillis()}")
+  }
+
+  def isAutoBump(pullRequest: PullRequestDraft, labels: List[Label]): Boolean = {
+    pullRequest.head.exists(_.ref.startsWith("trickle/")) && labels.exists(_.name == AutoBumpLabel)
+  }
+}
+
+class AutoBump(repository: OutdatedRepository, token: String) {
+  import AutoBump._
+
+  assert(url(repository.url).getHost == "github.com")
+
+  implicit private val IOContextShift: ContextShift[IO] = contextShift(global)
+  val github: Github[IO] = Github[IO](Some(token))
+  val (owner, repoSlug) = repository.ownerAndRepository.getOrElse(sys.error(s"invalid url ${repository.url}"))
+  val sbt: String = sys.env.getOrElse("SBT", "sbt")
+
+  /**
+   * Finds primary open autobump pull request, if one exists.
+   *
+   * @return Oldest autobump pull request, by order of creation
+   */
+  def getOldestAutoBumpPullRequest: IO[Option[PullRequestDraft]] = {
+    getPullRequests
+      .evalFilter(pullRequest => getLabels(pullRequest.number).map(labels => isAutoBump(pullRequest, labels)))
+      .head
+      .compile.toList
+      .map(_.headOption)
+  }
+
+  // TODO: use Last-Modified/If-Modified-Since
+  def getPullRequests: Stream[IO, PullRequestDraft] = {
+    autoPage(Pagination(1, 100)) { pagination =>
+      github.pullRequests.listDraftPullRequests(owner, repoSlug, PullRequestFilters, Some(pagination))
+    }
+  }
+
+  // FIXME: current github4s api does not support Some(pagination)
+  def getLabels(pr: Int): IO[List[Label]] = {
+    github.issues.listLabels(owner, repoSlug, pr).rethrow.map(_.result)
+  }
+
+  def draftPullRequest(authorRepository: String, branchName: String, changes: String): IO[PullRequestDraft] = {
+    github
+      .pullRequests
+      .draftPullRequest(
+        owner,
+        repoSlug,
+        NewPullRequestData(autoBumpCommitTitle(authorRepository),
```

Comment:
Style nit: put `autoBumpCommitTitle(authorRepository)` on the next line. Basically if you're wrapping actuals, either wrap all or none.